astropy/table/table.py

```diff
@@ -1903,6 +1903,52 @@ def remove_rows(self, row_specifier):
         if hasattr(self, '_groups'):
             del self._groups
 
+    def remove_na(self):
+        '''
+        Remove rows from a table that has N/A.
+        For masked tables only
+
+        Parameters
+        ----------
+        Masked Table
+
+        Examples
+        --------
+        Create a table with three columns 'a', 'b' and 'c'::
+
+            >>> weather_data = """
+            ...      day,temp,type
+            ...      ,35,rainy
+            ...      Tues,55,sunny
+            ...      Wed,31,snowy
+            ...      Thu,25,snowy
+            ...      Sun,1.1,
+            ...      """
+
+            >>> t = ascii.read(weather_data)
+            >>> print (t)
+            day  temp  type
+            ---- ---- -----
+              -- 35.0 rainy
+            Tues 55.0 sunny
+             Wed 31.0 snowy
+             Thu 25.0 snowy
+             Sun  1.1    --
+
+        Remove rows that have n/a values::
+
+            >>> t.remove_na()
+            >>> print(t)
+            day  temp  type
+            ---- ---- -----
+            Tues 55.0 sunny
+             Wed 31.0 snowy
+             Thu 25.0 snowy
+        '''
+
+        self.remove_rows(np.where([col.data for col in self.mask.itercols()])[-1])
```

Comment:
The implementation from https://stackoverflow.com/questions/50256012/drop-rows-with-masked-elements-in-astropy-table is going to be much more efficient:
```
keep_mask = reduce(operator.and_, [~col.mask for col in dat.itercols()])
```

From here I would actually suggest to improve (in this PR) the `remove_rows()` method to allow passing in a boolean mask.  So in `remove_rows()` you would check on the input and if it is a boolean mask then bypass the code which sets `keep_mask` in that method.

Finally, then just call `self.remove_rows(keep_mask)` within `remove_masked_rows()`.

Reply:
I think I tried to do what you have suggested and the test cases have passed. Lemme know if it looks alright.